File: hermesapi/hermesbeebop_client.py
import requests

host = 'http://0.0.0.0:8080'


def g(action, *args):
    url = host+'/'+action
    for arg in args:
        url += '/'+str(arg)
    r = requests.get(url)
    return r.json()


def HermesBeebop___init__():
    return g('HermesBeebop___init__', )['return_value']


def HermesBeebop_connect():
    return g('HermesBeebop_connect', )['return_value']


def HermesBeebop_disconnect():
    return g('HermesBeebop_disconnect', )['return_value']


def HermesBeebop_take_off():
    return g('HermesBeebop_take_off', )['return_value']


def HermesBeebop_land():
    return g('HermesBeebop_land', )['return_value']


def HermesBeebop_emergency():
    return g('HermesBeebop_emergency', )['return_value']


def HermesBeebop_flat_trim():
    return g('HermesBeebop_flat_trim', )['return_value']


def HermesBeebop_navigate_home():
    return g('HermesBeebop_navigate_home', )['return_value']


def HermesBeebop_move_by(dx, dy, dz, dr):
    return g('HermesBeebop_move_by', dx, dy, dz, dr)['return_value']


def HermesBeebop_max_altitude(m):
    return g('HermesBeebop_max_altitude', m)['return_value']


def HermesBeebop_max_tilt(d):
    return g('HermesBeebop_max_tilt', d)['return_value']


def HermesBeebop_max_distance(m):
    return g('HermesBeebop_max_distance', m)['return_value']


def HermesBeebop_autonomous_flight_max_horizontal_speed(ms):
    return g('HermesBeebop_autonomous_flight_max_horizontal_speed', ms)['return_value']


def HermesBeebop_autonomous_flight_max_vertical_speed(ms):
    return g('HermesBeebop_autonomous_flight_max_vertical_speed', ms)['return_value']


def HermesBeebop_autonomous_flight_max_horizontal_acceleration(mss):
    return g('HermesBeebop_autonomous_flight_max_horizontal_acceleration', mss)['return_value']


def HermesBeebop_autonomous_flight_max_vertical_acceleration(mss):
    return g('HermesBeebop_autonomous_flight_max_vertical_acceleration', mss)['return_value']


def HermesBeebop_autonomous_flight_max_rotation_speed(ms):
    return g('HermesBeebop_autonomous_flight_max_rotation_speed', ms)['return_value']


def HermesBeebop_max_vertical_speed(ms):
    return g('HermesBeebop_max_vertical_speed', ms)['return_value']


def HermesBeebop_max_rotation_speed(ms):
    return g('HermesBeebop_max_rotation_speed', ms)['return_value']


def HermesBeebop_set_home(lat, lon, alt):
    return g('HermesBeebop_set_home', lat, lon, alt)['return_value']


def HermesBeebop_get_location():
    return g('HermesBeebop_get_location', )['return_value']


def HermesBeebop_get_state():
    return g('HermesBeebop_get_state', )['return_value']


if __name__ == '__main__':
    pass

File: hermesapi/hermesbeebop_client.js
var request = require('request');

var clientApi = {
    host : 'http://0.0.0.0:8080',
    g : function(action, args){
        if(arguments.length < 2) {
            return {};
        } else {
            var options = {
                'url': this.host,
                'method': 'GET'
            };
            var callback = arguments[arguments.length - 1];
            for(var i = 0; i < arguments.length - 1; i++) {
                options['url'] += '/' + arguments[i];
            }
            request(options, function(err, res, body) {
                callback(err, res, body);
            });
        }
    },
    HermesBeebop_take_off: function( cb){
        this.g('HermesBeebop_take_off', cb);
    },
    HermesBeebop_land: function( cb){
        this.g('HermesBeebop_land', cb);
    },
    HermesBeebop_emergency: function( cb){
        this.g('HermesBeebop_emergency', cb);
    },
    HermesBeebop_flat_trim: function( cb){
        this.g('HermesBeebop_flat_trim', cb);
    },
    HermesBeebop_navigate_home: function( cb){
        this.g('HermesBeebop_navigate_home', cb);
    },
    HermesBeebop_move_by: function(dx, dy, dz, dr, cb){
        this.g('HermesBeebop_move_by', dx, dy, dz, dr, cb);
    },
    HermesBeebop_max_altitude: function(m, cb){
        this.g('HermesBeebop_max_altitude', m, cb);
    },
    HermesBeebop_max_tilt: function(d, cb){
        this.g('HermesBeebop_max_tilt', d, cb);
    },
    HermesBeebop_max_distance: function(m, cb){
        this.g('HermesBeebop_max_distance', m, cb);
    },
    HermesBeebop_autonomous_flight_max_horizontal_speed: function(ms, cb){
        this.g('HermesBeebop_autonomous_flight_max_horizontal_speed', ms, cb);
    },
    HermesBeebop_autonomous_flight_max_vertical_speed: function(ms, cb){
        this.g('HermesBeebop_autonomous_flight_max_vertical_speed', ms, cb);
    },
    HermesBeebop_autonomous_flight_max_horizontal_acceleration: function(mss, cb){
        this.g('HermesBeebop_autonomous_flight_max_horizontal_acceleration', mss, cb);
    },
    HermesBeebop_autonomous_flight_max_vertical_acceleration: function(mss, cb){
        this.g('HermesBeebop_autonomous_flight_max_vertical_acceleration', mss, cb);
    },
    HermesBeebop_autonomous_flight_max_rotation_speed: function(ms, cb){
        this.g('HermesBeebop_autonomous_flight_max_rotation_speed', ms, cb);
    },
    HermesBeebop_max_vertical_speed: function(ms, cb){
        this.g('HermesBeebop_max_vertical_speed', ms, cb);
    },
    HermesBeebop_max_rotation_speed: function(ms, cb){
        this.g('HermesBeebop_max_rotation_speed', ms, cb);
    },
    HermesBeebop_set_home: function(lat, lon, alt, cb){
        this.g('HermesBeebop_set_home', lat, lon, alt, cb);
    },
    HermesBeebop_get_location: function( cb){
        this.g('HermesBeebop_get_location', cb);
    },
    HermesBeebop_get_state: function( cb){
        this.g('HermesBeebop_get_state', cb);
    }
};
